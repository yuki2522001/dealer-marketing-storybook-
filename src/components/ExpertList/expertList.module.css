.expert-list {
  display: flex;
  margin: 0 auto;
  max-width: 1140px;
  justify-content: space-between;
  background-color: var(--light);
}
