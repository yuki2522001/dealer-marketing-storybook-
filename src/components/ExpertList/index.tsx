import React, { memo, useContext } from "react";
import Error from "next/error";
import CardExpert from "../CardExpert";
import styleExpertList from "./expertList.module.css";
import { ExpertContext } from "@context/ExpertContext";

const ExpertList: React.FC = () => {
  const { experts, errorCode } = useContext(ExpertContext);

  if (errorCode) {
    return <Error statusCode={errorCode} />;
  }

  return (
    <div data-testid="expert-list" className={styleExpertList["expert-list"]}>
      {experts?.map((expert) => (
        <div key={expert.expertId}>
          <CardExpert expert={expert} />
        </div>
      ))}
    </div>
  );
};

export default memo(ExpertList);
