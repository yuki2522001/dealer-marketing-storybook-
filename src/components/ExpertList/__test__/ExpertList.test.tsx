import { render } from "@testing-library/react";
import ExpertList from "../index";

describe("ExpertList component", () => {
  test("Should render ExpertList component", () => {
    const { getByTestId } = render(<ExpertList />);
    const expertList = getByTestId("expert-list");
    expect(expertList).toBeInTheDocument();
  });

  test("Should render ExpertList component when error data", () => {
    const { getByTestId } = render(<ExpertList />);
    const expertList = getByTestId("expert-list");
    expect(expertList).toBeInTheDocument();
  });

  test("Matches snapshot", () => {
    const { container } = render(<ExpertList />);
    expect(container).toMatchSnapshot();
  });
});
