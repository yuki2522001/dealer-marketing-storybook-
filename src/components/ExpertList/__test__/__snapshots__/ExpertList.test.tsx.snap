// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpertList component Matches snapshot 1`] = `
<div>
  <div
    data-testid="expert-list"
  />
</div>
`;
